### ISSUE TITLE
Setuptools version requirement is too strict.

### ISSUE THREAD

**Issue by jettisonjoe** [pantsbuild/pex#390]
From version.py:
```Python
# NB: If we upgrade to setuptools>=34 pex's bootstrap code in `PEXBuilder` will need an update to
# include the `packaging` package in the `.bootstrap/` code since we use
# `packaging.specifiers.SpecifierSet` - indirectly - through `pkg_resources.Requirement.specifier`.
SETUPTOOLS_REQUIREMENT = 'setuptools>=20.0,<34.0'
```

Modern Python installations include setuptools version numbers at 35 or beyond, so using the PEX API (at least on a directory that includes a setup.py) now requires its own separate Python environment with a "valid" version of setuptools installed, extending the rabbit hole of building and distributing a package yet one level deeper. Is there a fix planned?

**Comment by nmost** [comment]
I was able to monkey patch locally within our code base to get pex to run without any visible down sides: https://github.com/pantsbuild/pex/compare/master...nmost:version-bump-setuptools

**Comment by jsirois** [comment]
This has been solved since Pex 1.6.0 with vendoring of all Pex dependencies in #624.
